.github/workflows/lint.yml

```
@@ -0,0 +1,73 @@
+name: Lint
+on:
+  workflow_call:
+
+jobs:
+  lint_go:
+    name: Lint Go 🔍
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code 🛎️
+      uses: actions/checkout@v4
+
+    - name: Set up Go 🏗️
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: go.mod
+        cache: false
+
+    - name: Lint alloy module 🔍
+      uses: golangci/golangci-lint-action@v6
+      with:
+        version: v1.64.7
+
+    - name: Lint syntax module 🔍
+      uses: golangci/golangci-lint-action@v6
+      with:
+        version: v1.64.7
+        working-directory: ./syntax/
+
+    - name: Run custom linters 🔍
+      run: make run-alloylint
+
+  shellcheck:
+    name: Shellcheck 🔍
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code 🛎️
+      uses: actions/checkout@v4
+
+    - name: Shellcheck 🔍
+      uses: azohra/shell-linter@latest
```

```suggestion
      uses: Azbagheri/shell-linter@latest
```
Seems the repo was [migrated](https://github.com/Azbagheri/shell-linter?tab=readme-ov-file#-repository-transferred--manual-update-required).

---

I'll update this! I'll pin a hash for v0.8.0 as well.